### ISSUE TITLE
Clarify the set of default flags we apply for codegen

### ISSUE THREAD

**Issue by lmazuel** [Azure/autorest.csharp#2142]
Part of pushing DPG to be the default story, we wil need to clarify what should be the default behavior of autorest and what flags can trigger different behaviors.

Right now I think we can genreate 3 flavors of C# code:
- Code for HLC. Intended to be wrapped. Include models.
- Code for DPG.
- Code for mgmt

I'd like to see first a reply to this issue
- What is the default behavior today if I do "autorest --csharp"
- What are the flags used for the 3 flavors today. 

For the GA, we want to mark GA as a milestone by changing the default set of flags, but I'd like to make a recommendation only once I saw that current status.

**Comment by lmazuel** [comment]
Closing in favor of https://github.com/azure/autorest.csharp/issues/1769
